build/sync/plan/checks.go

```diff
@@ -99,41 +99,67 @@ func (r PathExistsChecker) Check(path string, ctx Setup) error {
 // reference will default to the source repository and repo - to the target.
 type FileUnalteredChecker struct {
 	Params struct {
-		ReferenceRepo RepoID `json:"compared-to"`
-		Repo          RepoID `json:"in"`
+		SourceRepo RepoID `json:"compared-to"`
+		TargetRepo RepoID `json:"in"`
 	}
 }
 
 // Check implements the Checker interface.
 func (f FileUnalteredChecker) Check(path string, setup Setup) error {
 	setup.Logf("checking if file %q has not been altered", path)
-	repo := f.Params.Repo
+	repo := f.Params.TargetRepo
 	if repo == "" {
 		repo = TargetRepo
 	}
-	reference := f.Params.ReferenceRepo
-	if reference == "" {
-		reference = SourceRepo
+	source := f.Params.SourceRepo
+	if source == "" {
+		source = SourceRepo
 	}
-	absPath := setup.PathInRepo(repo, path)
+	trgPath := setup.PathInRepo(repo, path)
+	srcPath := setup.PathInRepo(source, path)
 
-	info, err := os.Stat(absPath)
-	if os.IsNotExist(err) {
-		return CheckFailf("file %q has been deleted", absPath)
-	}
+	fileHashes, err := git.FileHistory(path, setup.GetRepo(source).Git)
 	if err != nil {
-		return fmt.Errorf("failed to get stat for %q: %v", absPath, err)
+		return err
 	}
-	if info.IsDir() {
-		return fmt.Errorf("%q is a directory", absPath)
+
+	var srcDeleted bool
+	srcInfo, err := os.Stat(srcPath)
+	if os.IsNotExist(err) {
+		srcDeleted = true
+	} else if err != nil {
+		return fmt.Errorf("failed to get stat for %q: %v", trgPath, err)
+	}
```

Comment:
Is this more readable? 

```suggestion
	if err != nil {
		if os.IsNotExist(err) {
			srcDeleted = true
		} else {
			return fmt.Errorf("failed to get stat for %q: %v", trgPath, err)
		}
	}
```

Reply:
I'm fine with both variants.